src/shared/bg_pmove.cpp

```diff
@@ -4516,6 +4516,21 @@ PmoveSingle
 
 ================
 */
+// set the firing flag for continuous beam weapons
+static void setFireBeam( buttonNumber_t btn )
+{
+	if ( !( pm->ps->pm_flags & PMF_RESPAWNED ) && pm->ps->pm_type != PM_INTERMISSION &&
+			usercmdButtonPressed( pm->cmd.buttons, BTN_ATTACK ) &&
+			( ( pm->ps->ammo > 0 || pm->ps->clips > 0 ) || BG_Weapon( pm->ps->weapon )->infiniteAmmo ) )
+	{
+		pm->ps->eFlags |= EF_FIRING;
+	}
+	else
+	{
+		pm->ps->eFlags &= ~EF_FIRING;
```

Comment:
the previous code did set `EF_FIRING2` or `EF_FIRING3` in some cases, too. I would use that as second argument.

Reply:
good catch, again.